api/resolver/init.go

```diff
@@ -17,7 +17,12 @@ import (
 // The Container should be used like regular [NNSResolver].
 type Container struct {
 	mu       sync.RWMutex
-	resolver *NNSResolver
+	resolver NNSResolver
+}
+
+// NNSResolver allows to resolve container id by its name.
+type NNSResolver interface {
```

Comment:
Interface with the only implementation in the same package can't exist.

Reply:
Sure, it was too much